src/snowcli/cli/common/snow_cli_global_context.py

```diff
@@ -0,0 +1,41 @@
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Callable
+
+
+@dataclass
+class LoggingAndExceptionHandlingContext:
+    enable_tracebacks: bool
+
+
+@dataclass
+class SnowCliGlobalContext:
+    logging_and_exception_handling: LoggingAndExceptionHandlingContext
+
+
+class SnowCliGlobalContextHolder:
```

Comment:
```suggestion
class SnowCliGlobalContextManager:
```
WDYT? I've never meet `holder` concept, is this a common practice?

Reply:
Manager is a good name. I'll change it.

Reply:
As we agreed to not use the solution from this PR, I made changes from this comment in new PR: https://github.com/Snowflake-Labs/snowcli/pull/192